spec/rule_spec.rb

```diff
@@ -3,6 +3,18 @@
 require 'spec_helper'
 
 describe RRule::Rule do
+  describe '#has_ended?' do
+    it 'returns true when there are no more events' do
+      rrule = RRule.parse 'DTSTART:20210805T122300Z;FREQ=DAILY;UNTIL=20210818T121000Z'
```

Comment:
I don't think this gem supports specifying `DTSTART` like this. That it doesn't blow up feels like a bug. I think all these specs pass because `dtstart: Time.now` is past the `UNITL`s, so all these rules are empty.

Reply:
>I don't think this gem supports specifying DTSTART like this


In our project we're implementing this npm package for the frontend:  https://www.npmjs.com/package/rrule
[Example](https://jakubroztocil.github.io/rrule/)